### PR TITLE
[utils] Fix MONO_SOLIB_EXT in mono-dl.h on Apple ARM64 (now .dylib)

### PR DESCRIPTION
**Pull request by k15tfu**

### CODE REVIEW

**Comment by k15tfu** [comment]
cc @vargaz

**Review by vargaz — APPROVED** [review]

File: mono/utils/mono-dl.h [hunk 1]
```diff
@@ -17,6 +17,8 @@
 #define MONO_SOLIB_EXT ".dylib"
 #elif defined(TARGET_MACH) && defined(TARGET_AMD64)
 #define MONO_SOLIB_EXT ".dylib"
+#elif defined(TARGET_MACH) && defined(TARGET_ARM64)
+#define MONO_SOLIB_EXT ".dylib"
```
Comment (vargaz) [inline mono/utils/mono-dl.h:21]: Could probably be #ifdef TARGET_MACH.

Comment (k15tfu) [reply mono/utils/mono-dl.h:15]: Will do.

**Comment by marek-safar** [comment]
@monojenkins approve

**Comment by lambdageek** [comment]
Remaining failures are unrelated

**Pull request merged**